src/lib/transaction_logic/zkapp_command_logic.ml

```diff
@@ -834,7 +834,19 @@ module Make (Inputs : Inputs_intf) = struct
     Stack_frame.make ~caller:default_caller ~caller_caller:default_caller
       ~calls:(Call_forest.empty ())
 
-  let assert_ = Bool.Assert.is_true
+  let assert_ b pos =
+    try Bool.Assert.is_true b with
```

Comment:
Can you move this logic into the corresponding implementations in `Transaction_logic` and `Transaction_snark`?